src/signed.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{Index, KeyBox};
+use codec::{Decode, Encode};
+
+pub trait Signable {
+    fn bytes_to_sign(&self) -> Vec<u8>;
+}
+
+#[derive(Clone, Debug, Decode, Encode)]
+pub struct UncheckedSigned<T: Signable + Index, S> {
+    pub signable: T,
+    pub signature: S,
+}
+
+#[derive(Debug)]
+pub struct Signed<'a, T: Signable + Index, KB: KeyBox> {
+    pub unchecked: UncheckedSigned<T, KB::Signature>,
+    pub key_box: &'a KB,
+}
+
+impl<'a, T: Signable + Index + Clone, KB: KeyBox> Clone for Signed<'a, T, KB> {
+    fn clone(&self) -> Self {
+        Signed {
+            unchecked: self.unchecked.clone(),
+            key_box: self.key_box,
+        }
+    }
+}
+
+impl<'a, T: Signable + Index, KB: KeyBox> Signed<'a, T, KB> {
+    pub fn sign(key_box: &'a KB, signable: T) -> Self {
+        let signature = key_box.sign(&signable.bytes_to_sign());
+        let signed = signable;
```

Comment:
Why this temp var is created?

Reply:
The idea was that semantically `signable` is no longer just an instance of the `Signable`, but some something which is signed. But now that I look at it i agree that it does not seem reasonable given how simple the function is.